crates/burn-jit/src/kernel/reduce/naive/shader.rs

```diff
@@ -1,148 +1,42 @@
-use cubecl::{
-    cpa,
-    ir::{Elem, KernelDefinition, Scope, Variable, Visibility},
-    CubeCountSettings, Execution, InputInfo, KernelExpansion, KernelIntegrator, KernelSettings,
-    OutputInfo,
-};
-use std::marker::PhantomData;
-
-use crate::{element::JitElement, kernel::Kernel, tensor::JitTensor, JitRuntime};
+use crate::{element::JitElement, tensor::JitTensor, JitRuntime};
+use cubecl::calculate_cube_count_elemwise;
+use cubecl::prelude::*;
 
 use super::base::ReduceDimNaive;
 
-pub(crate) struct NaiveReduceDimComputeShader<E: JitElement, RD: ReduceDimNaive<E>> {
-    tensor: Variable,
-    dim: usize,
-    output: Variable,
-    _reduce_dim: PhantomData<RD>,
-    _elem: PhantomData<E>,
-}
-
-#[derive(new)]
-pub(crate) struct NaiveReduceDimEagerKernel<
-    RD: ReduceDimNaive<EI>,
-    R: JitRuntime,
-    EI: JitElement,
-    EO: JitElement,
-> {
-    dim: usize,
-    reduce_dim: PhantomData<RD>,
-    _runtime: PhantomData<R>,
-    _elem_in: PhantomData<EI>,
-    _elem_out: PhantomData<EO>,
-}
-
-impl<RD: ReduceDimNaive<EI>, R: JitRuntime, EI: JitElement, EO: JitElement> Kernel
-    for NaiveReduceDimEagerKernel<RD, R, EI, EO>
-{
-    fn define(&self) -> KernelDefinition {
-        let mut scope = Scope::root();
-        let item_input = EI::cube_elem().into();
-        let item_output = EO::cube_elem().into();
+#[cube(launch)]
+pub(crate) fn naive_reduce_dim_compute_shader<RD: ReduceDimNaive<EI>, EI: Numeric, EO: Numeric>(
+    input: &Tensor<EI>,
+    output: &mut Tensor<EO>,
+    dim: UInt,
+) {
+    if ABSOLUTE_POS >= output.len() {
+        return;
+    }
```

Comment:
If you are doing bound checks, you can change `#[cube(launch)]` to `#[cube(launch_unchecked)`] and call `launch_unchecked` for improved performance.

Reply:
done!